src/includes/init.php

```diff
@@ -19,9 +19,11 @@ function path($file)
 
 function authorize($role)
 {
-    if (!isset($_SESSION) || $_SESSION['role'] != $role) {
+    if (!isset($_SESSION) || !isset($_SESSION['role']) || $_SESSION['role'] != $role) {
+        $_SESSION['errtitle'] = 'Unauthorized user';
+        $_SESSION['errcode'] = "Unauthorized user";
+        redirect("/includes/errorPage/errorPage.php");
         http_response_code(401);
```

Comment:
This will not work for a couple of reasons:
1) `redirect` exits after setting location header to prevent the page from rendering any potentially sensitive data, so this line is never reached.
2) HTTP code needs to be 302 for redirects.

`http_response_code($xyz);` will need to be moved to `errorPage.php` instead.

Reply:
moved to error page